src/BatteryStateOfChargeService.h

```diff
@@ -19,4 +22,12 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     const double initialStateOfChargePercent_;
 
     // Recommend adding necessary variables here
+
+    QTime previousTime;
+    QTime newTime;
+    double previousCurrent;
+    double deltaTime;
+    double aveCurrent;
+
```

Comment:
Names in the header should follow the format "someName_", for example, previousTime should be previousTime_